src/httpfs.cpp

```diff
@@ -842,7 +884,7 @@ void HTTPFileHandle::Initialize(optional_ptr<FileOpener> opener) {
 				etag = value.etag;
 
 				if (flags.OpenForReading()) {
-					read_buffer = duckdb::unique_ptr<data_t[]>(new data_t[READ_BUFFER_LEN]);
+					read_buffer = AllocateReadBuffer(opener);
```

Comment:
I think this (possibly in a follow up) can be completely skipped for `flags.DirectIO() ||flags.RequireParallelAccess()` (both here and below) since then we will skip the buffer.

That should avoid a bunch of allocations, that are likely not the bottleneck, but should just be better.

Or maybe better: `AllocateReadBuffer` should take in consideration the flags, and potentially avoid allocating at all

Reply:
Good idea. Added this to the PR